runtime/lua/vim/shared.lua

```diff
@@ -1139,4 +1139,24 @@ function vim._defer_require(root, mod)
   })
 end
 
+--- @private
+function vim._with(context, f)
+  -- vim._with({ options = { isfname = vim.o.isfname .. ',@-@' } }, function() end)
+  local previous_options = {}
+  if context.options then
```

Comment:
Minor nit but I would only assign `previous_options` if `context.options` is defined. Then the condition on line 1155 can change to `if previous_options then`.

This avoids having to allocate the table if `context.options` isn't passed.